### PR TITLE
dapp: Refactor to use smart meter API calls + bring back Eve and draw energy consumption graph

### PR DESCRIPTION
**Pull request by Norbo11**
The energy consumption graph is now drawn, using real data fetched from the smart meter Eve API.

### CODE REVIEW

**Comment by Norbo11** [comment]
Don't merge this yet

File: client/meter/blockchain.js [hunk 1]
```diff
@@ -32,9 +32,10 @@ async function mySellerContracts() {
     const seller = await contract.seller();
     if (seller === account) {
       const offeredAmount = await contract.offeredAmount();
-      if (offeredAmount == 0) continue;
+      if (offeredAmount == 0)
+        continue;
       const unitPrice = await contract.unitPrice();
-      contracts.push({contractAddr, unitPrice, offeredAmount})
+      contracts.push({contractAddr: contractAddr, unitPrice: unitPrice, offeredAmount: offeredAmount})
```
Comment (fangyi-zhou) [inline client/meter/blockchain.js]: Why?

Comment (Norbo11) [reply client/meter/blockchain.js]: So that I can access the fields by their name?

Comment (fangyi-zhou) [reply client/meter/blockchain.js]: http://www.benmvp.com/learning-es6-enhanced-object-literals/

Comment (Norbo11) [reply client/meter/blockchain.js]: Oh right, I forgot about those. Will change

**Comment by fangyi-zhou** [comment]
Please rebase master and cherry pick commits related to this PR

File: dapp/src/client/scripts/app.jsx [hunk 2]
```diff
@@ -26,92 +26,79 @@ export class App extends React.Component {
...
-      const remainingEnergyInContract = await contract.remainingEnergy(this.state.account);
-      const seller = await contract.seller();
-      const offeredAmount = await contract.offeredAmount();
-      const unitPrice = await contract.unitPrice();
-      if (this.state.account === seller) {
-        if (offeredAmount > 0) {
+  async getAvailableContracts() {
+    fetch('http://localhost:3000/contract/available_contracts')
+      .then(response => response.json())
+      .then(contracts => contracts.filter((c) => c.offeredAmount > 0))
```
Comment (fangyi-zhou) [inline dapp/src/client/scripts/app.jsx]: You shouldn't need to do that i suppose

File: dapp/src/client/scripts/app.jsx [hunk 3]
```diff
@@ -26,92 +26,79 @@ export class App extends React.Component {
...
           }))
-        } else {
+        })
+      })
+  }
+
+  async getSellerContracts() {
+    fetch('http://localhost:3000/contract/my_seller_contracts')
+      .then(response => response.json())
+      .then(contracts => contracts.filter((c) => c.offeredAmount > 0))
```
Comment (fangyi-zhou) [inline dapp/src/client/scripts/app.jsx]: ditto

**Comment by fangyi-zhou** [comment]
Closing

**Pull request closed without merging**